lib/matplotlib/figure.py

```diff
@@ -1637,526 +1826,959 @@ def _make_array(inp):
+    is confined to a prescribed region of the figure.  For example the
+    following puts two subfigures side-by-side::
+
+        fig = plt.figure()
+        sfigs = fig.subfigures(1, 2)
+        axsL = sfigs[0].subplots(1, 2)
+        axsR = sfigs[1].subplots(2, 1)
+
+    See :doc:`/gallery/subplots_axes_and_figures/subfigures`
+    """
+
+    def __init__(self, parent, subplotspec, *,
+                 facecolor=None,
+                 edgecolor=None,
+                 linewidth=0.0,
+                 frameon=None):
+        """
+        Parameters
+        ----------
+        parent : `.figure.Figure` or `.figure.SubFigure`
+            Figure or subfigure that contains the SubFigure.  SubFigures
+            can be nested.
+
+        subplotspec : `.gridspec.SubplotSpec`
+            defines the region in a parent gridspec where the subfigure will
+            be placed
+
+        facecolor : default: :rc:`figure.facecolor`
+            The figure patch facecolor.
+
+        edgecolor : default: :rc:`figure.edgecolor`
+            The figure patch edge color.
+
+        linewidth : float
+            The linewidth of the frame (i.e. the edge linewidth of the figure
+            patch).
+
+        frameon : boolean, default: :rc:`figure.frameon`
+            If ``False``, suppress drawing the figure background patch.
+        """
+        super().__init__()
+        if facecolor is None:
+            facecolor = mpl.rcParams['figure.facecolor']
+        if edgecolor is None:
+            edgecolor = mpl.rcParams['figure.edgecolor']
+        if frameon is None:
+            frameon = mpl.rcParams['figure.frameon']
+
+        # self.clf()
+        self._subplotspec = subplotspec
+        self._parent = parent
+        self.figure = parent.figure
+        # subfigures use the parent axstack
+        self._axstack = parent._axstack
+        self.subplotpars = parent.subplotpars
+        self.dpi_scale_trans = parent.dpi_scale_trans
+        self._axobservers = parent._axobservers
+        self.dpi = parent.dpi
+        self.canvas = parent.canvas
+        self.transFigure = self._parent.transFigure
+        self.bbox_relative = None
+        self._redo_transform_rel_fig()
+        self.figbbox = self._parent.figbbox
+        self.bbox = TransformedBbox(self.bbox_relative,
+                                    self._parent.transSubfigure)
+        self.transSubfigure = BboxTransformTo(self.bbox)
+
+        self.patch = Rectangle(
+            xy=(0, 0), width=1, height=1, visible=frameon,
+            facecolor=facecolor, edgecolor=edgecolor, linewidth=linewidth,
+            # Don't let the figure patch influence bbox calculation.
+            in_layout=False, transform=self.transSubfigure)
+        self.patch.set_antialiased(False)
+
+        if parent._layoutgrid is not None:
+            self.init_layoutgrid()
+
+    def _redo_transform_rel_fig(self, margins=(0, 0, 0, 0), bbox=None):
```

Comment:
Appears to never be called with `bbox` set to anything.

Reply:
This is called in `constrained_layout.py`, line 453...

Reply:
OTOH, it is the case that *margin* is never set to anything!  So I've removed that

Reply:
Ah sorry, I only checked in this file.